### PR TITLE
Fix issue in Reducible

### PR DESCRIPTION
**Pull request by yilinwei**
courtesy of @ceedubs finding the bug.

### CODE REVIEW

**Comment by ceedubs** [comment]
👍 LGTM but maybe this is a sign that we should have a unit test for this :)

**Comment by yilinwei** [comment]
@ceedubs point taken....

**Comment by kailuowang** [comment]
I think in principle every method that is not a simple delegation should be covered.  This [chrome extension](https://chrome.google.com/webstore/detail/codecov-extension/keefkhehidemnokodkdkejapdgfjmijf?hl=en-US) is really helpful in checking that.

**Comment by peterneyens** [comment]
I think the problem here is that all the instances of `Reducible` override `get`. Another method I can think of with the same issue is ` Foldable#size`.

It would be nice if we could somehow check the consistency (#1684) and benchmark (something we talked about in #1532, see https://github.com/typelevel/cats/pull/1532#issuecomment-27732622) overridden and the default implementation of type class methods.

**Comment by kailuowang** [comment]
How about we add a doctest to cover this and address the overrides test coverage in #1684? That way we can merge this fix sooner.

**Comment by peterneyens** [comment]
@kailuowang, do you mean something like : 

```scala
scala> import cats.data.NonEmptyList
scala> import cats.implicits._

scala> val R = new NonEmptyReducible[NonEmptyList, List] {
     |   def split[A](nel: NonEmptyList[A]): (A, List[A]) = (nel.head, nel.tail)
     | }

scala> val nel = NonEmptyList.of(1,2,3)

scala> R.get(nel, 1L)
res0: Option[Int] = Some(2)

scala> R.get(nel, 4L)
res1: Option[Int] = None
```

**Comment by kailuowang** [comment]
@peterneyens yup. that would work.

File: core/src/main/scala/cats/Reducible.scala [hunk 1]
```diff
@@ -227,7 +227,7 @@ abstract class NonEmptyReducible[F[_], G[_]](implicit G: Foldable[G]) extends Re
   }
 
   override def get[A](fa: F[A])(idx: Long): Option[A] =
-    if (idx == 0L) Some(split(fa)._1) else G.get(split(fa)._2)(idx)
+    if (idx == 0L) Some(split(fa)._1) else G.get(split(fa)._2)(idx - 1L)
```
Comment (johnynek) [inline core/src/main/scala/cats/Reducible.scala]: this is not actually tested according to codecov.

Can we write a test to explicitly exercise this?

Comment (yilinwei) [inline core/src/main/scala/cats/Reducible.scala]: @johnynek done.

**Comment by yilinwei** [comment]
Failure in the test is due to the `Kleisli` `Eq` presumably not working as intended; given that this PR doesn't touch that are we good to merge?

**Comment by djspiewak** [comment]
Merging with two sign-offs.

**Pull request merged**